src/python/WMCore/MicroService/DataStructs/MSOutputTemplate.py

```diff
@@ -0,0 +1,256 @@
+    """
+    A simple template to represent the objects stored in Mongodb for MSOutput
+    Its purpose should be to serve as a bidirectional representation of the
+    objects stored in Mongodb. The instances of this class should be used as a
+    buffer for both inbound (reading from the database) and outbound (writing to
+    database) documents. And it must have all the methods needed to set or update
+    various fields in the object. So that we assure the uniformity of all documents.
+
+    Template format:
+    {
+    "RequestName": "ReqName",
+    "RequestStatus": "(clompletd|cloed-out|announced),
+    'Campaign': "Campaign",
+    "creationTime": integer timestamp,
+    "lastUpdate": integer timestamp,
+    "isRelVal": (True|False),
+    "isTaken": (True|False),
+    "destination": ["list of locations"],
+    "OutputDatasets": ["list of datasets"],
+    "destinationOutputMap": [{"destination": ["list of locations"],
+                              "datasets": ["list of datasets"]},
+                             {"destination": ["list of locations"],
+                              "datasets": ["list of datasets"]}],
+    "campaignOutputMap": [{"campaignName": "blah",
+                           "datasets": ["list of datasets"]},
+                          {"campaignName": "blah",
+                           "datasets": ["list of datasets"]}],
+    "transferStatus": "blah", # either "pending" or "done",
+    "transferIDs": ["list of transfer IDs"],
+    "numberOfCopies": integer
+    }
+    """
+    # TODO:
+    #    To add an identifier parameter to __init__
+    #    to distinguish inbound from outbound documents.
+
+    # DONE:
+    #    to figure out how to pass params as dict not only kwargs so I can use
+    #    the same logic for both inbound and outbound documents
+
+    # TODO:
+    #    to implement document mutex in a model similar to the one described here:
+    #    https://stackoverflow.com/questions/32728670/mutex-with-mongodb
+    #    ```
+    #    db.collection.update({done: false, taken: false},{$set: {taken: true, takenBy: myIp}});
+    #    ```
+
+    # TODO:
+    #    as soon as any change in the instantiated object happens ( any call of
+    #    a set method) the same change must be updated in Mongodb too -
+    #    this may happen from outside or from the object itself. The former method
+    #    keeps the document independent from the underlying Database technology
+
+    def __init__(self, doc=None, **kwargs):
+        super(MSOutputTemplate, self).__init__(**kwargs)
+        # super(MSOutputTemplate, self).__init__()
+
+        docTemplate = [
+            ('_id', None, unicode),
+            ('RequestName', None, unicode),
+            ('Campaign', None, unicode),
+            ('creationTime', None, Timestamp),
+            ('lastUpdate', None, Timestamp),
+            ('isRelVal', None, bool),
+            ('isTaken', False, bool),
+            ('isTakenby', None, (str, unicode)),
```

Comment:
Perhaps we should be consistent with these keys, so:
* "isTakenby" --> "isTakenBy"
* for the others, always start with a non-capital letter.

Reply:
then reflect the same changes in the docstring above

Reply:
> "isTakenby" --> "isTakenBy"
This I will fix. 

> for the others, always start with a non-capital letter.

It will take me out of consistency with the same keys coming from Reqmgr, because there they are all starting with Capital letters. And if I do so, I will need to implement additional logic to sort/search/map them while fetching them from Reqmgr Api and trying to modify them  to the current naming schema. While  right now it happens naturally because of this: [1]
I would rather say lets make them all starting with Capital letter here too.

[1]
https://github.com/dmwm/WMCore/blob/22297f258696f5d0704fc46517b4db2319d10e3f/src/python/WMCore/MicroService/DataStructs/MSOutputTemplate.py#L106-L108

Reply:
> "isTakenby" --> "isTakenBy"

Done

Reply:
>  I would rather say lets make them all starting with Capital letter here too.

I agree! Please start all of them with capital letters then.